### PR TITLE
Fix bugs

### PR DESCRIPTION
**Pull request by ChamseddineBhd**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

**What is the current behavior?** *(You can also link to an open issue here)*

**What is the new behavior (if this is a feature change)?**

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: src/components/network/network.js [hunk 1]
```diff
@@ -37,6 +37,8 @@ export default class Network {
 
     lines = [];
 
+    voltageLevels = [];
```
Comment (geofjamg) [inline src/components/network/network.js]: It is redundant with voltageLevelsById

File: src/components/single-line-diagram.js [hunk 2]
```diff
@@ -45,7 +45,7 @@ const SingleLineDiagram = (props) => {
             var bbox = svg.getBBox();
             svg.setAttribute("viewBox", 0 + " " + 0 + " " + (bbox.width + 20) + " " + (bbox.height + 20));
         }
-    }, []);
+    }, [props]);
```
Comment (geofjamg) [inline src/components/single-line-diagram.js]: You could be more precise by just adding properties (just property.svg?) that need to reapply the hack

File: src/redux/reducer.js [hunk 3]
```diff
@@ -44,4 +47,6 @@ export const reducer = createReducer(initialState, {
     [REMOVE_VOLTAGE_LEVEL_DIAGRAM]: (state, action) => {
         state.diagram = null;
     }
-});
\ No newline at end of file
+});
+
+export const store = createStore(reducer);
```
Comment (geofjamg) [inline src/redux/reducer.js]: Whyt have you move store creation here. This is strange to put store creation in reducer file.

File: src/components/single-line-diagram.js (hunk 2 above)
Comment (ChamseddineBhd) [reply src/components/single-line-diagram.js]: ok

File: src/redux/reducer.js (hunk 3 above)
Comment (ChamseddineBhd) [reply src/redux/reducer.js]: ok i'll move it in a seprate file but it should be in the redux folder. what do you think ?

File: src/components/network/network.js [hunk 4]
```diff
@@ -47,13 +49,16 @@ export default class Network {
 
     setSubstations(substations) {
         this.substations = substations;
+        console.log(substations)
```
Comment (geofjamg) [inline src/components/network/network.js]: this is for debugging => to remove

File: src/components/network/network.js [hunk 5]
```diff
@@ -90,7 +95,7 @@ export default class Network {
     }
 
     getVoltageLevels() {
-        return Array.from(this.substations.values());
+        return Array.from(this.voltageLevels);
```
Comment (geofjamg) [inline src/components/network/network.js]: You can get this using voltageLevelsById values

File: src/redux/reducer.js (hunk 3 above)
Comment (geofjamg) [reply src/redux/reducer.js]: ok

**Pull request merged**